EIPS/eip-6960.md

```diff
@@ -0,0 +1,328 @@
+     * @param mainId is the main token type ID being transferred
+     * @param subId subId is the token subtype ID being transferred
+     * @param amount is the amount of tokens being transferred
+     * @param data is additional data with no specified format
+     * @return `IDLTReceiver.onDLTReceived.selector`
+     */
+    function onDLTReceived(
+        address operator,
+        address from,
+        uint256 mainId,
+        uint256 subId,
+        uint256 amount,
+        bytes calldata data
+    ) external returns (bytes4);
+}
+```
+
+## Rationale
+
+The two-level classification system introduced in this EIP allows for a more organized token ecosystem, enabling users to manage and track tokens with greater granularity. It is particularly useful for projects that require token classifications beyond the capabilities of the current ERC-1155 standard.
+
+As assets can have various properties or variations, our smart contract design reflects this by assigning a mainId to each asset category and a unique subId to each derivative or sub-category. This approach expands the capabilities of ERC-1155 to support a broader range of assets with complex requirements. Additionally, it enables tracking of mainBalance for the main asset and subBalance for its sub-assets individual accounts.
+
+The contract can be extended to support the use of subIds in two ways:
+
+- Shared SubIds: where all mainIds share the same set of subIds.
+- Mixed SubIds: where mainIds have unique sets of subIds.
+
+DLT provides a more versatile solution compared to other token standards such as ERC-20, ERC-721, and ERC-1155 by effectively managing both fungible and non-fungible assets within the same contract.
+
+The following are questions that we considered during the design process:
+
+- How to name the proposal?
+The standard introduces a two-level classification to tokens where one main asset (layer 1) can be further sub-divided into several sub-assets (layer 2) hence we decided to name it as "Dual-layer" token to reflect the hierarchical structure of the token classification.
+- Should we limit the classification to two levels?
+The standard’s implementation maintains a mapping to track the total supply of each sub-asset. If we allow sub-assets to have their own children, it would be necessary to introduce additional methods to track each sub-asset, which would be impractical and increases the complexity of the contract.
+- Should we extend the ERC-1155 standard?
+As the ERC-1155 standard is not designed to support a layered classification and requires significant modifications to do so, we concluded that it would not be appropriate to extend it for the dual-layer token standard. Hence, a standalone implementation would be a more suitable approach.
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Test Cases
+
+The standard is implemented and tested with full coverage on Polytrade's Dual-Layer-Token repository
```

Comment:
Please remove this external link (even if it isn't an HTML/markdown link :wink:)

Reply:
Sorry if I wasn't clear. Mentioning your repository for tests cases or a reference implementation isn't permitted, as it violates our external link policy. Either include your test cases and implementation in the EIP itself, or in the assets directory.

Reply:
My bad, I forgot about that one. Thanks for reminding me. It's all fixed now!